x-pack/solutions/observability/plugins/slo/server/services/purge_rollup_data.test.ts

```diff
@@ -0,0 +1,192 @@
+
+    it('successfully makes a query to remove SLI data older than a week', async () => {
+      const slo = createSLO({
+        id: 'test2',
+        indicator: createAPMTransactionErrorRateIndicator(),
+        timeWindow: {
+          type: 'calendarAligned',
+          duration: new Duration(1, DurationUnit.Week),
+        },
+      });
+      mockRepository.findAllByIds.mockResolvedValueOnce([slo]);
+
+      await purgeRollupData.execute({
+        list: ['test2'],
+        purgePolicy: { purgeType: 'fixed_age', age: new Duration(2, DurationUnit.Week) },
+      });
+
+      expect(mockRepository.findAllByIds).toMatchSnapshot();
+      expect(mockEsClient.deleteByQuery).toMatchSnapshot();
+    });
+
+    it('successfully makes a query to remove SLI data based on a timestamp - month', async () => {
+      const slo = createSLO({
+        id: 'test3',
+        indicator: createAPMTransactionErrorRateIndicator(),
+        timeWindow: {
+          type: 'calendarAligned',
+          duration: new Duration(1, DurationUnit.Month),
+        },
+      });
+      mockRepository.findAllByIds.mockResolvedValueOnce([slo]);
+
+      await purgeRollupData.execute({
+        list: ['test3'],
+        purgePolicy: { purgeType: 'fixed_time', timestamp: new Date('2025-03-01T00:00:00Z') },
+      });
+
+      expect(mockEsClient.deleteByQuery).toMatchSnapshot();
+    });
+
+    it('successfully makes a query to remove SLI data based on a timestamp - week', async () => {
+      const slo = createSLO({
+        id: 'test4',
+        indicator: createAPMTransactionErrorRateIndicator(),
+        timeWindow: {
+          type: 'calendarAligned',
+          duration: new Duration(1, DurationUnit.Week),
+        },
+      });
+      mockRepository.findAllByIds.mockResolvedValueOnce([slo]);
+
+      await purgeRollupData.execute({
+        list: ['test4'],
+        purgePolicy: { purgeType: 'fixed_time', timestamp: new Date('2025-04-01T00:00:00Z') },
+      });
+
+      expect(mockEsClient.deleteByQuery).toMatchSnapshot();
+    });
+
+    it('successfully makes a forced query to remove recently added SLI data', async () => {
+      const slo = createSLO({
+        id: 'test2',
+        indicator: createAPMTransactionErrorRateIndicator(),
+        timeWindow: {
+          type: 'calendarAligned',
+          duration: new Duration(1, DurationUnit.Week),
+        },
+      });
+      mockRepository.findAllByIds.mockResolvedValueOnce([slo]);
+
+      await purgeRollupData.execute({
+        list: ['test2'],
+        purgePolicy: {
+          purgeType: 'fixed_age',
+          age: new Duration(1, DurationUnit.Day),
+        },
+        force: true,
+      });
+
+      expect(mockEsClient.deleteByQuery).toMatchSnapshot();
+    });
+  });
+
+  describe('error path', () => {
+    it('fails to make a query to remove SLI data older than 7 days', async () => {
+      const slo = createSLO({
+        id: 'test1',
+        indicator: createAPMTransactionErrorRateIndicator(),
```

Comment:
same comment, define the timeWindow in all your crafted SLOs, keep the default indicator